astropy/coordinates/baseframe.py

```diff
@@ -945,6 +945,32 @@ def is_frame_attr_default(self, attrnm):
         """
         return attrnm in self._attr_names_with_defaults
 
+    def is_equivalent_frame(self, other):
+        """
+        Checks if this object is the same frame as the ``other`` object.
+
+        To be the same frame, two objects must be the same frame class and have
+        the same frame attributes.  Note that it does *not* matter what, if any,
+        data either object has.
+
+        Parameters
+        ----------
+        other : BaseCoordinateFrame
+            the other frame to check
+
+        Returns
+        -------
+        isequiv : bool
+            True if the frames are the same, False if not.
+        """
+        if self.__class__ == other.__class__:
+            for frame_attr_name in self.get_frame_attr_names():
```

Comment:
As purely a style thing, I like the functional approach when possible, so would have written:
```
isequiv = (self.__class__ is other.__class__ and
           all(getattr(self, fan) == getattr(other, fan) for fan in self.get_frame_attr_names()))
return isequiv
```
But it can be a matter of taste and gives the same result.